g1ant-language/desktop-apps-automation2.md

```diff
@@ -0,0 +1,211 @@
+## Unsubscribe Me: Mails, Text Tools And File Operations
+
+The previous exercise with G1ANT.Language was based on Excel and the Robot commands designed for this application (our language also provides specialized commands for Word and Outlook). You learned how to use procedures for clean code and repetitive tasks. Now you can try some more commands to automate desktop apps.
+
+Imagine such a scenario: you run a company, which sends email newsletters to its subscribers. The subscribers list is stored in an ordinary CSV (comma separated values) file — it’s a plain text file with basic information: first name, last name, company name and email address of a subscriber. When your subscriber no longer wants to receive emails from you, he or she sends an email with “unsubscribe” in the subject field and you manually delete this address from your mailing list.
+
+![Sample mailing list CSV file in Notepad](../-assets/mailing-list.jpg)
+
+And here the G1ANT.Robot steps in and takes this job from you!
+
+Start with designing an architecture for your script: the plan of actions.
+
+1. Login into your email account.
+2. Check for unread emails containing “unsubscribe” in the subject field.
+3. If there are any, store the sender’s email address.
+4. Open the mailing list .csv file.
+5. Search for the stored email addresses and delete them from the list.
+6. Save and close the file.
+
+Seems pretty easy. And, in fact, it is with our G1ANT.Language commands.
+
+### Silent Execution
+
+You could do all of the above actions using UI commands you learned in the [Basics](../g1ant-language/basics.md) chapter: `program`, `window` and `keyboard`. They would open your email client (or webmail in a browser), search for “unsubscribe” message subject, copy the sender’s email, then open the mailing list file in Notepad and use its Find and Replace tool to delete the address from the list. But there are less obtrusive ways that do not involve dealing with the user interface and clicking around menus, dialogs and so on. They work directly with the target objects in the background, so users can still do tasks on their PCs.
```

Comment:
don't mistake general descriptions for our tools. "UI commands" means "user interface commands" -- that is, commands that interact with UI. If it's not clear to you, I can expand the abbreviation in the text. But UI is a common abbreviation, so don't worry.

Reply:
Because you stubbornly try to insist that "UI commands" always refers to a precise set of commands in an addon. No, "UI commands" means exactly what it means. And again no, most of the commands are not leveraging the UI as a user does -- at least not in a way that is as explicit as with the program, keyboard or window commands.
OK, let's finish this fruitless discussion. I will expand UI.

Reply:
```suggestion
You could do all of the above actions with the user interface commands you learned in the [Basics](../g1ant-language/basics.md) chapter: `program`, `window` and `keyboard`. They would open your email client (or webmail in a browser), search for “unsubscribe” message subject, copy the sender’s email, then open the mailing list file in Notepad and use its Find and Replace tool to delete the address from the list. But there are less obtrusive ways that do not involve dealing with the user interface and clicking around menus, dialogs and so on. They work directly with the target objects in the background, so users can still do tasks on their PCs.
```

Reply:
I will repeat what I'm saying "I am only saying that it is misleading because then 95% of our commands are ui". So let me explain that in detail. I don’t see a difference between the commands that you brought in the conversation and commands like excel, outlook, image, chrome, word, selenium, ie, color all of them do something that is visible on our screens, operate on user interface. So no, there isn’t anything like “in between”, either a command operates on ui or it does not. And yes, most of the commands are ui, this what RPA is about - robots follow exactly step by step what a user does on their computer. 
I will not bring this topic again but if you use "ui commands" expression, then maybe you should explain what you are talking about. If your manual is for dumb people as you always tell, then it should be easy to understand for everyone. G1ANT.Addon.UI is one of our important addons and even though we have so many bugs there and there is no command described in the manual, people use it and ask about it.